cmd/celestia/rpc.go

```diff
@@ -0,0 +1,165 @@
+package main
+
+import (
+	"bytes"
+	"encoding/base64"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/spf13/cobra"
+)
+
+var requestURL string
+var authTokenFlag string
+
+type jsonRPCRequest struct {
+	ID      int64         `json:"id"`
+	JSONRPC string        `json:"jsonrpc"`
+	Method  string        `json:"method"`
+	Params  []interface{} `json:"params"`
+}
+
+func init() {
+	rpcCmd.PersistentFlags().StringVar(
+		&requestURL,
+		"url",
+		"http://localhost:26658",
+		"Request URL",
+	)
+	rpcCmd.PersistentFlags().StringVar(
+		&authTokenFlag,
+		"auth",
+		"",
+		"Authorization token (if not provided, the AUTH_TOKEN environment variable will be used)",
```

Comment:
[non-blocking] I'd update this to `CELESTIA_NODE_AUTH_TOKEN` as `AUTH_TOKEN` is pretty generic for a environment variable name.

Reply:
agreed